changelogs/fragments/9238-locale-gen-rewrite.yml

```
@@ -0,0 +1,14 @@
+minor_changes:
+  - "locale_gen - invert the logic to determine ``ubuntu_mode``, making it look first for ``/etc/locale.gen`` (set ``ubuntu_mode`` to ``False``)
+    and only then looking for ``/var/lib/locales/supported.d/`` (set ``ubuntu_mode`` to ``True``) (
+    https://github.com/ansible-collections/community.general/pull/9238,
+    https://github.com/ansible-collections/community.general/issues/9131,
+    https://github.com/ansible-collections/community.general/issues/8487
+    )."
```

Otherwise you have spaces between `(` and `)` and the content inbetween.
```suggestion
    and only then looking for ``/var/lib/locales/supported.d/`` (set ``ubuntu_mode`` to ``True``)
    (https://github.com/ansible-collections/community.general/pull/9238,
    https://github.com/ansible-collections/community.general/issues/9131,
    https://github.com/ansible-collections/community.general/issues/8487)."
```